### PR TITLE
ignore empty and comment lines in gitallowed

### PR DESCRIPTION
**Pull request by michaelwittig**
ignore empty and comment lines in gitallowed

### CODE REVIEW

**Comment by mtdowling** [comment]
This is cool thanks. Would you be able to add a test for skipping comments?

**Comment by michaelwittig** [comment]
I added a test and also updated the docs. 

Btw: We run your tool now in production (~50 developer) for two weeks. With good results: Teams use it and we have results.

File: git-secrets [hunk 1]
```diff
@@ -54,10 +54,10 @@ load_patterns() {
 }
 
 load_allowed() {
-  git config --get-all secrets.allowed
+  git config --get-all secrets.allowed | awk 'NF && $1!~/^#/'
```
Comment (mtdowling) [inline git-secrets]: I don't think we need this filtering on git config values -- they can't contain comments, and I don't think they can be empty. Is there a need to filter here?

**Comment by michaelwittig** [comment]
@mtdowling It now only ignores empty lines and comments in .gitallowed

**Comment by mtdowling** [comment]
The only remaining question I have around this is whether or not this would prohibit adding a pattern that starts with `#` to the .gitallowed file. If it does preclude that, do you think that's a reasonable tradeoff?

**Comment by michaelwittig** [comment]
@mtdowling We saw a few people building .gitallowed files like the following (we have a pattern trying to search for api keys (uuids). So far we have no pattern that looks for something that contains a # sign. 

```
# uuids for testing
550e8400-e29b-11d4-a716-446655440000

# commit hashes for dependencies
9a034128a329f4fb7a53043dd1d1e8f74bfc91fc
```

My personal opinion: People are used to comments so it feels natural to support them. But that's not a strong argument...

**Comment by mtdowling** [comment]
Yeah, I think starting a pattern with `#` is an edge case. The workaround there would be to just add it in your git-config instead of through the .gitallowed. I'm happy with this change.

**Pull request merged**